workspaces/arborist/lib/arborist/index.js

```diff
@@ -35,6 +35,55 @@ const { log, time } = require('proc-log')
 const { saveTypeMap } = require('../add-rm-pkg-deps.js')
 const AuditReport = require('../audit-report.js')
 const relpath = require('../relpath.js')
+const { LRUCache } = require('lru-cache')
+
+class PackumentCache extends LRUCache {
+  static #heapLimit = require('node:v8').getHeapStatistics().heap_size_limit
+
+  #sizeKey
+  #disposed = new Set()
+
+  #log (...args) {
+    // It's all silly
+    log.silly('packumentCache', ...args)
+  }
+
+  constructor ({ heapFactor = 0.25, maxEntryFactor = 0.5, sizeKey = '_contentLength' } = {}) {
+    const maxSize = Math.floor(PackumentCache.#heapLimit * heapFactor)
+    const maxEntrySize = maxSize * maxEntryFactor
+    super({
+      maxSize,
+      maxEntrySize,
+      // Don't cache if we dont know the size
+      sizeCalculation: (p) => p[sizeKey] || maxEntrySize + 1,
```

Comment:
We can now tweak this based on the findings of memory usage to size correlation.

Under 10k = .45
Under 1M = .6

max entry size should floor at 1M, above that the tradeoff is not worth the memory.